### PR TITLE
feat: allow the configuration of the GKE release channel

### LINKED ISSUE #98: Expose ability to specify kubernetes version or release channel for GKE Clusters
There doesn't exist to be a method to specify a version of the kubernetes or release channel for GKE clusters in this module. Allowing the user to specify this would be beneficial.

### PR DESCRIPTION
**Pull request by garethjevans**
fixes #98

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -119,6 +119,7 @@ The following two paragraphs provide the full list of configuration and output v
 | node\_disk\_size | Node disk size in GB | `string` | `"100"` | no |
 | node\_machine\_type | Node type for the Kubernetes cluster | `string` | `"n1-standard-2"` | no |
 | parent\_domain | The parent domain to be allocated to the cluster | `string` | `""` | no |
+| release\_channel | GKE [Release Channel](https://cloud.google.com/kubernetes-engine/docs/concepts/release-channels) to subscribe to. | `string` | `REGULAR` | no |
```
Comment (hferentschik) [inline README.md:122]: See https://github.com/jenkins-x/terraform-google-jx/issues/98#issuecomment-638335968

How does this relate to specifying an actual Kubernetes version?

File: modules/cluster/variables.tf [hunk 2]
```diff
@@ -78,6 +78,12 @@ variable "max_node_count" {
   default     = 5
 }
 
+variable "release_channel" {
+  description = "The GKE release channel to subscribe to.  See https://cloud.google.com/kubernetes-engine/docs/concepts/release-channels"
+  type        = string
+  default     = "REGULAR"
```
Comment (abayer) [inline modules/cluster/variables.tf:84]: Feels like this should probably be `STABLE` by default, since `REGULAR` is currently on 1.16.

**Comment by hferentschik** [comment]
Can we squash this into a single commit?

File: modules/cluster/main.tf [hunk 3]
```diff
@@ -72,7 +76,7 @@ resource "google_container_node_pool" "jx_node_pool" {
 
   management {
     auto_repair  = "true"
-    auto_upgrade = "false"
+    auto_upgrade = var.release_channel != "UNSPECIFIED"
```
Comment (hferentschik) [inline modules/cluster/main.tf:79]: 👍

**Pull request merged**